src/systems/hydrodynamics/Hydrodynamics.cc

```diff
@@ -340,13 +344,16 @@ void Hydrodynamics::PreUpdate(
   state(4) = localRotationalVelocity.Y();
   state(5) = localRotationalVelocity.Z();
 
-  auto dt = static_cast<double>(_info.dt.count())/1e9;
-  stateDot = (state - this->dataPtr->prevState)/dt;
+  auto dt = (double)_info.dt.count()/1e9;
```

Comment:
Don't use a c-style cast here. Also it might be worth using an incantation like: `auto dt = std::chrono::duration<double>(info.dt).count();` (I'm not sure if its correct but `std::chrono` has magical functions) insteat of hard coding` 1e9`.